test/v1alpha1/service.go

```diff
@@ -344,3 +393,194 @@ func IsServiceRoutesNotReady(s *v1alpha1.Service) (bool, error) {
 	result := s.Status.GetCondition(v1alpha1.ServiceConditionRoutesReady)
 	return s.Generation == s.Status.ObservedGeneration && result != nil && result.Status == corev1.ConditionFalse, nil
 }
+
+// RestoreGateway updates the gateway object to the oldGateway
+func RestoreGateway(t *testing.T, clients *test.Clients, oldGateway v1alpha3.Gateway) {
+	curGateway, err := clients.SharedClient.NetworkingV1alpha3().Gateways(Namespace).Get(GatewayName, metav1.GetOptions{})
+	if err != nil {
+		t.Fatalf("Failed to get Gateway %s/%s", Namespace, GatewayName)
+	}
+	curGateway.Spec.Servers = oldGateway.Spec.Servers
```

Comment:
nit: we probably want to check if the Spec.Servers are the same btw currGateway and oldGateway. If they are the same, we don't need update.